src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -137,10 +138,21 @@
 ;; request body will also have the configuration required to connect and lookup
 ;; for secrets from the external Secret Manager.
 (defmethod handle-request "go.cd.secrets.secrets-lookup"
-  [_ _ data]
-  ;; TODO: See https://plugin-api.gocd.org/19.7.0/secrets/#lookup-secrets for desired response
-  (let [configuration (:configuration data)
-        secret-keys (:keys data)]
-    {:response-code    500
-     :response-headers {}
-     :response-body    "NYI"}))
+  [client _ data]
+  (try
+    (let [secret-keys (:keys data)
+          secrets (mapv (fn [key]
+                          {:key   key
+                           :value (vault/read-secret client key)})
+                        secret-keys)]
+      {:response-code    200
+       :response-headers {}
+       :response-body    secrets})
+    (catch ExceptionInfo ex
+      (if (= "No such secret" (first (str/split (ex-message ex) #":")))
```

Comment:
I'd still really like to see this behavior changed, but a less complicated way to accomplish this would be:
```suggestion
      (if (str/starts-with? (ex-message ex) "No such secret:")
```

Reply:
Fixing now :)